docs/README.rst

```diff
@@ -26,7 +26,7 @@ the following tools are needed to build the documentation:
 In a conda environment, or a Python 3 ``venv``, you should be able to run::
 
   cd ipython
-  pip install .[doc] -U
+  pip install -U -r docs/requirements.txt
```

Comment:
I think this file is here just for readthedocs, user should likely not use it.
Though they may need quotes. Maybe:

```suggestion
  pip install --upgrade '.[doc]' 
```

Reply:
As I note in af5f1156ee8fbff4e30d6f81f5f6f5ff08673d89, if one follows the instructions in `docs/README.rst`, and then to build the API docs, makes changes, and then builds the docs again, the new changes do not show up in the docs build because the package is not installed in editable mode. That is why I think using the `requirements.txt` is better, since it has `-e`.